src/foam/nanos/crunch/ServerCrunchService.java

```diff
@@ -753,5 +759,25 @@ private Predicate getAssociationPredicate_(X x) {
         EQ(AgentCapabilityJunction.EFFECTIVE_USER, user.getId())
       )
     );
+    // Consider the capability that is being referenced,
+    // and assign correct predicate
+    if ( capabilityId != null ) {
+      DAO capabilityDAO = (DAO) x.get("capabilityDAO");
+      Capability cap = (Capability) capabilityDAO.find(capabilityId);
+      if ( cap == null ) return result;
```

Comment:
maybe this should be an error instead of returning result

Reply:
lgtm otherwise

Reply:
If you are trying to find a ucj for a capability that can't be found ... log an error? throw an exception or ignore. Ruby, you suggest a good point ... let me look